index.bs

```diff
@@ -357,10 +355,46 @@ execute the following steps:
 5.  [$Set the `Sec-CH-UA` header for a request$], given |r|.
 
 
-IDL {#idl} 
-===
+Interface {#interface} 
+=================
+
+<pre class="idl">
+[Exposed=Window]
+interface NavigatorUAData {
+  readonly attribute DOMString brand;
+  readonly attribute DOMString version;
+  readonly attribute DOMString platform;
+  readonly attribute DOMString architecture;
+  readonly attribute DOMString model;
+  readonly attribute boolean mobile;
+};
+interface mixin NavigatorUA {
+  [SecureContext] Promise&lt;NavigatorUAData&gt; getUserAgent();
+};
+Navigator includes NavigatorUA;
+
+</pre>
 
-ISSUE: Yoav should write all this.
+Processing model {#processing}
+--------------
+<dfn method for="NavigatorUA"><code>getUserAgent()</code></dfn> method *must*
+run these steps:
+
+1. Let |UAData| be a new {{NavigatorUAData}} object.
+2. Set |UAData|'s {{NavigatorUAData/brand}} to the value of the user agent's [=user agent/brand=].
+3. Set |UAData|'s {{NavigatorUAData/platform}} to the value of the user agent's
+    [=user agent/platform brand and version=].
+4. Set |UAData|'s {{NavigatorUAData/architecture}} to the value of the user
+    agent's [=user agent/platform architecture=].
+5. Set |UAData|'s {{NavigatorUAData/model}} to the value of the user agent's [=user agent/model=].
+6. Set |UAData|'s {{NavigatorUAData/mobile}} to the value of the user agent's
+    [=user agent/mobileness=].
+7. Let |version| be the value of the user agent's [=user agent/major version=].
+8. Let |settings| be the [=context object=]’s [=relevant settings object=].
+9. If |settings|'s [=environment settings object/client-hints set=] [=list/contains=] `UA`, then
+    set |version| to be the value of the user agent's [=user agent/full version=].
+10. Set |UAData|'s {{NavigatorUAData/version}} to the value of |version|.
+11. Return |UAData|.
```

Comment:
We need to return a promise that resolves to |UAData| "in parallel".

Reply:
Yup!